File: Python/PROGRAMMERS/HASH/완주하지못한선수.py
def solution(participant, completion):
    p_dict = {}
    for p in participant:
        if p_dict.get(p):
            p_dict[p] += 1
        else:
            p_dict[p] = 1
    for c in completion:
        if p_dict.get(c):
            p_dict[c]-=1
    print()
    return sorted(list(p_dict.items()),key=lambda x:x[1])[-1][0]

File: JavaScript/PROGRAMMERS/완주하지못한선수.js
function solution(participant, completion) {
    var answer = '';
    const temp = {}
    for (let i=0;i<participant.length;i++){
        if (temp[participant[i]]) {
            temp[participant[i]] += 1
        }
        else {
            temp[participant[i]] = 1
        }
    }
    for (let i=0;i<completion.length;i++) {
        temp[completion[i]] -= 1
    }
    for (let t in temp) {
        if (temp[t]){
            answer = t
            break
        }
    }
    return answer;
}
var participant = ["marina", "josipa", "nikola", "vinko", "filipa"]
var completion = ["josipa", "filipa", "marina", "nikola"]
solution(participant,completion)